python/cudf/cudf/testing/csv.py

```diff
@@ -0,0 +1,184 @@
+import copy
+import json
+import logging
+import os
+import random
+import sys
+
+import cudf
+from cudf.testing.utils import _generate_rand_meta
+from cudf.tests import dataset_generator as dg
+
+logging.basicConfig(
+    format="%(asctime)s %(levelname)-8s %(message)s",
+    level=logging.INFO,
+    datefmt="%Y-%m-%d %H:%M:%S",
+)
+
+
+class CSVReader(object):
+    def __init__(
+        self,
+        file_name="temp_csv.csv",
+        dirs=None,
+        max_rows=4096,
```

Comment:
This seems pretty small, no?

Reply:
I had it kept at a smaller default value locally but passed in `max_rows` param value through `pythonfuzz` decorator while running the tests. Any suggestions on what the default has to be?

Reply:
I guess if we're always overriding it it isn't a big deal. Another thought could be to default to `None` and then pick a random number in order to add further test cases to the fuzzing tests.

Reply:
I haven't committed the overriding changes for all functions so that we(devs) could be able to run these on their workstations.

If we default to `None`, what would the fallback random number be for the size of rows?

Reply:
Instead of 100_000, what if we generated a random number between 0 and 100_000? That way we get further test coverage for small corner / edge cases as well?

Reply:
Yea, `max_rows` is used for that purpose. The resulting dataframes are of size varying from 0 to `max_rows`.

Reply:
That logic is here: https://github.com/rapidsai/cudf/pull/6114/files#diff-baf658dc5cb5bdf7f82b1fee4d5a7ef7R8